From d636ef8cfefdaa564e1363373adc77d80529e0c3 Mon Sep 17 00:00:00 2001
From: Alex Stewart <alex.stewart@ni.com>
Date: Sat, 5 Nov 2022 16:56:59 -0500
Subject: [PATCH] semaphore.h: fix undefined condition_variable type

Building the grpc server component on Debian 11 and NILRT 10 throws an
error in semaphore.h about `std::condition_variable` being undefined.
g++ recommends explicitly adding the `condition_variable` header.

Add the header.

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

Upstream-Status: Submitted [https://github.com/ni/grpc-device/pull/786]

---
 source/server/semaphore.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/source/server/semaphore.h b/source/server/semaphore.h
index bd25461..ac8411f 100644
--- a/source/server/semaphore.h
+++ b/source/server/semaphore.h
@@ -2,6 +2,7 @@
 #define NIDEVICE_GRPC_SEMAPHORE_H
 
 #include <shared_mutex>
+#include <condition_variable>
 
 namespace nidevice_grpc {
 
-- 
2.38.1

